src/data/store.rs

```diff
@@ -79,13 +80,37 @@ impl Store {
         store.downloads.load_downloaded_files();
         store.downloads.load_pending_downloads();
 
+        store.chats.load_chats();
+
         store.search.load_featured_models();
         store
     }
 
-    pub fn load_model(&mut self, file: &File) {
-        if self.chats.load_model(file).is_ok() {
-            self.preferences.set_current_chat_model(file.id.clone());
+    // TODO: This wrappers feel weird.
+    pub fn create_new_chat(&mut self, file: &File, set_as_current: bool) {
+        if self.chats.create_new_chat(file, set_as_current).is_ok() {
+            // TODO: This breaks, FIX THE PREFERENCES!!!
+            // self.preferences.set_current_chat_model(file.id.clone());
+        }
+    }
+
+    /// This loads the model
+    // TODO: Feels weird to do this wrappers
```

Comment:
I would agree that simple wrappers do not make sense. But here we have the chance to have more than a wrapper.

As I said in another comment, this function should be responsible for getting the `available_model_files` (no real need to receive it from the caller) and call the `chats.set_current_chat` functions.